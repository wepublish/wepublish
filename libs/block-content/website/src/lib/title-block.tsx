import {styled} from '@mui/material'
import {Block, TitleBlock as TitleBlockType} from '@wepublish/website/api'
import {BuilderTitleBlockProps, useWebsiteBuilder} from '@wepublish/website/builder'

export const isTitleBlock = (block: Block): block is TitleBlockType =>
  block.__typename === 'TitleBlock'

export const TitleBlockWrapper = styled('div')`
  display: grid;
  gap: ${({theme}) => theme.spacing(5)};
`

export const TitleBlock = ({title, lead, className}: BuilderTitleBlockProps) => {
  const {
    elements: {H3, H6}
  } = useWebsiteBuilder()

  return (
    <TitleBlockWrapper className={className}>
      <H3 component="h1">{title}</H3>

      <H6 component="p">{lead}</H6>
    </TitleBlockWrapper>
  )
}
