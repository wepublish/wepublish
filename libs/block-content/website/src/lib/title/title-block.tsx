import styled from '@emotion/styled'
import {BlockContent, TitleBlock as TitleBlockType} from '@wepublish/website/api'
import {BuilderTitleBlockProps, useWebsiteBuilder} from '@wepublish/website/builder'
import {H3} from '@wepublish/ui'
import {useReadingList} from '@wepublish/reading-list/website'

export const isTitleBlock = (block: Pick<BlockContent, '__typename'>): block is TitleBlockType =>
  block.__typename === 'TitleBlock'

export const TitleBlockWrapper = styled('div')`
  display: grid;
  gap: ${({theme}) => theme.spacing(3)};
  grid-auto-rows: min-content;
`
export const TitleBlockTitle = styled(H3)``

export const TitleBlock = ({title, lead, className}: BuilderTitleBlockProps) => {
  const {
    elements: {H6}
  } = useWebsiteBuilder()
  const [readingListProps] = useReadingList()

  return (
    <TitleBlockWrapper className={className} {...readingListProps}>
      <TitleBlockTitle component="h1">{title}</TitleBlockTitle>
      {lead && <H6 component="p">{lead}</H6>}
    </TitleBlockWrapper>
  )
}
