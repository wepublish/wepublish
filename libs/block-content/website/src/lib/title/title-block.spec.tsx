import {render} from '@testing-library/react'
import * as stories from './title-block.stories'
import {composeStories} from '@storybook/react'

const storiesCmp = composeStories(stories)

describe('Title Block', () => {
  Object.entries(storiesCmp).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      const {asFragment} = render(<Component />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
