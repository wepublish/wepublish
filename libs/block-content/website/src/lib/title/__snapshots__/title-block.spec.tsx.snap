// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Block should render Default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  gap: 24px;
  grid-auto-rows: min-content;
}

.emotion-2 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 600;
  line-height: 1.15;
  font-size: 3rem;
}

.emotion-3 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 300;
  font-size: 1.25rem;
  line-height: 1.4;
}

<div
    class="emotion-0 emotion-1"
  >
    <h1
      class="MuiTypography-root MuiTypography-h3 emotion-2"
    >
      This is a title
    </h1>
    <p
      class="MuiTypography-root MuiTypography-h6 emotion-3"
    >
      This is a lead
    </p>
  </div>
</DocumentFragment>
`;

exports[`Title Block should render WithClassName 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  gap: 24px;
  grid-auto-rows: min-content;
}

.emotion-2 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 600;
  line-height: 1.15;
  font-size: 3rem;
}

.emotion-3 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 300;
  font-size: 1.25rem;
  line-height: 1.4;
}

<div
    class="extra-classname emotion-0 emotion-1"
  >
    <h1
      class="MuiTypography-root MuiTypography-h3 emotion-2"
    >
      This is a title
    </h1>
    <p
      class="MuiTypography-root MuiTypography-h6 emotion-3"
    >
      This is a lead
    </p>
  </div>
</DocumentFragment>
`;

exports[`Title Block should render WithEmotion 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  gap: 24px;
  grid-auto-rows: min-content;
}

.emotion-2 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 600;
  line-height: 1.15;
  font-size: 3rem;
}

.emotion-3 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 300;
  font-size: 1.25rem;
  line-height: 1.4;
}

<div
    class="emotion-0 emotion-1"
  >
    <h1
      class="MuiTypography-root MuiTypography-h3 emotion-2"
    >
      This is a title
    </h1>
    <p
      class="MuiTypography-root MuiTypography-h6 emotion-3"
    >
      This is a lead
    </p>
  </div>
</DocumentFragment>
`;

exports[`Title Block should render WithoutLead 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  gap: 24px;
  grid-auto-rows: min-content;
}

.emotion-2 {
  margin: 0;
  font-family: Hanken Grotesk,Roboto,sans-serif;
  font-weight: 600;
  line-height: 1.15;
  font-size: 3rem;
}

<div
    class="emotion-0 emotion-1"
  >
    <h1
      class="MuiTypography-root MuiTypography-h3 emotion-2"
    >
      This is a title
    </h1>
  </div>
</DocumentFragment>
`;
